.github/workflows/pathogen-repo-build.yaml

```diff
@@ -144,6 +144,23 @@ on:
           .snakemake/log/
         required: false
 
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        description: AWS access key ID, required for aws-batch runtime only
+        required: false
+
+      AWS_SECRET_ACCESS_KEY:
+        description: AWS secret access key, required for aws-batch runtime only
+        required: false
+
+      GITHUB_PAT:
+        description: GitHub personal access token, required for additional permissions such as dispatching cross repo workflows
+        required: false
+
+      SLACK_TOKEN:
+        description: Slack Bot User OAuth Token, required for Slack notifications only
+        required: false
```

Comment:
I think it's clearer to list all of the expected secrets but I agree it would be a hassle if we ever want to change the secret names or add/remove secrets. (I keep waffling on this 😅 )

Another issue is I don't think we should pass _all_ inherited secrets to the build runtime. I would add a `runtime-secrets` input (or something similar) to allow calling workflows to specify which secrets to pass to the build runtime. This would still require maintaining a list of secret names in the calling workflows.

Reply:
> I agree it would be a hassle if we ever want to change the secret names or add/remove secrets.

Yeah, we're definitely going to want to do that. One of the goals of centralizing is that it _shouldn't_ be a hassle to update things like that.

> Another issue is I don't think we should pass all inherited secrets to the build runtime. I would add a `runtime-secrets` input (or something similar) to allow calling workflows to specify which secrets to pass to the build runtime. This would still require maintaining a list of secret names in the calling workflows.

Agreed that we shouldn't pass all, but not agreed about making callers spell everything out. Using `secrets: inherit` gives us a single central place to modify what set of secrets is passed (i.e. this called workflow), and that's the primary value of it. I think most calling workflows will be fine with that. If a calling workflow that wants more control over it, it can not pass any secrets as secrets per se but instead pass everything it needs explicitly as env vars.

Thoughts?

Reply:
Hmm, so if a pathogen repo has a unique secret that needs to be passed to the build runtime, it would require us to update the reusable workflow?

> If a calling workflow that wants more control over it, it can not pass any secrets as secrets per se but instead pass everything it needs explicitly as env vars.

I don't think this is possible...The `with` key does not have access to the `secrets` context so it would force the calling workflow to pass the raw values of the secrets.

Reply:
> Hmm, so if a pathogen repo has a unique secret that needs to be passed to the build runtime, it would require us to update the reusable workflow?

I think we should _avoid_ that. But doesn't the explicit list of secrets (as in this PR right now) mandate that? (Am I missing something?)

> I don't think this is possible...The `with` key does not have access to the `secrets` context so it would force the calling workflow to pass the raw values of the secrets.

Oh! Bah. Well, there are alternatives with similar properties… but I'm starting to think there's too much recapitulating of the `nextstrain build` interface in the GitHub Actions workflow inputs here, and maybe we should just be letting the calling workflow specify args directly. That's already a centralized interface we maintain backwards compat for; another layer of indirection is seeming not useful.

For example, with https://github.com/nextstrain/cli/pull/289 in mind, what about:

1. The caller uses `secrets: inherit` to avoid tight coupling between caller and called.
2. The reusable workflow makes all secrets available in the environment of the `nextstrain build` step, but _not_ the runtime environment.
3. The resusable workflow has an input for providing arguments to `nextstrain build` before the build dir, and the caller passes `--env SECRET_NAME` as needed to make select inherited secrets available in the runtime environment. 

The last bit would replace the current `env` input too, and maybe even the `cpus` and `memory` inputs. 

Happy to chat about this in a higher bandwidth setting than PR comments…

Reply:
A more concrete example of what I'm proposing, tweaked from the seasonal-flu workflow: https://gist.github.com/tsibley/66bad2e2dd89c39423f3401f5cc04e21

Or potentially even: https://gist.github.com/tsibley/8d6dfbe7f797e3584db5f31c3f3947ae

Thinking aloud here.

Reply:
For posterity, we discussed this separately and agreed that the layer of indirection through inputs is not helpful. 
It would be easier for callers to be able to directly provide/see the full `nextstrain build` command in a single input. This would simplify the transition from running builds locally to the automated GH Actions. 

The expected `nextstrain build` commands are also simplified now that we no longer need the `--exec` option to pass environment variable to the build.

Reply:
Updated in [latest push](https://github.com/nextstrain/.github/compare/3f05452c6894e9b3f9ff50fb4c94c10650f5433f...b79ec646e73f496a7639e609de5365993bb3ae54?expand=1). I would love to know if there's a better way to run the provided command besides using `eval`.

Reply:
[Followed up closer to the code in question](https://github.com/nextstrain/.github/pull/44#discussion_r1267317818)